pkg/version/version_test.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidateSemver(t *testing.T) {
+	err := validateSemver("UNKNOWN")
+	assert := assert.New(t)
+	assert.NotNil(err)
+	err = validateSemver("0.0.0-1-gdf6a1cc-dirty")
+	assert.Nil(err)
+}
```

Comment:
Yes that's what I meant/why I added the comment to the end after the dirty check :-)